core/domain/service.go

```diff
@@ -0,0 +1,163 @@
+package domain
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/raystack/frontier/core/authenticate"
+	"github.com/raystack/frontier/core/organization"
+	"github.com/raystack/frontier/core/user"
+	"github.com/raystack/frontier/internal/bootstrap/schema"
+)
+
+type UserService interface {
+	GetByID(ctx context.Context, id string) (user.User, error)
+}
+
+type OrgService interface {
+	ListByUser(ctx context.Context, userID string) ([]organization.Organization, error)
+	AddMember(ctx context.Context, orgID, relationName string, principal authenticate.Principal) error
+}
+
+type Service struct {
+	repository  Repository
+	userService UserService
+	orgService  OrgService
+}
+
+const (
+	DNSChallenge = "_frontier-challenge.%s"
```

Comment:
What if we set it as `_frontier-domain-challenge=%s`

Reply:
Have read about this format being used else where as well 
- https://letsencrypt.org/docs/challenge-types/#dns-01-challenge
- https://www.digitalocean.com/community/tutorials/how-to-acquire-a-let-s-encrypt-certificate-using-dns-validation-with-acme-dns-certbot-on-ubuntu-18-04#step-3-setting-up-acme-dns-certbot

Reply:
We are using TXT records as key value pairs, although I know letsencrypt use it as `_frontier-challenge.%s` but I think clear distinction between key and value should be better approach using `=` as a separator in `_frontier-domain-verification=%s`. Also we are making it specifically for domain verification by adding `domain` in the name.